cli/bpmetadata/types_ui_ext.go

```diff
@@ -0,0 +1,161 @@
+
+	// If true, custom machine types will not be selectable.
+	// More info:
+	// https://cloud.google.com/compute/docs/instances/creating-instance-with-custom-machine-type
+	DisallowCustomMachineTypes bool `yaml:"disallowCustomMachineTypes,omitempty"`
+}
+
+type GceGpuTypeExtension struct {
+	MachineType string `yaml:"machineType"`
+	GpuType     string `yaml:"gpuType"`
+}
+
+type GceGpuCountExtension struct {
+	// This field references another variable from the schema,
+	// which must have type GceMachineType
+	MachineTypeVariable string `yaml:"machineTypeVariable"`
+}
+
+type GceDiskSizeExtension struct {
+	// The allowable range of disk sizes depends on the disk type. This field
+	// references another variable from the schema, which must have type GceDiskType
+	DiskTypeVariable string `yaml:"diskTypeVariable"`
+}
+
+type GceNetworkExtension struct {
+	// AllowSharedVpcs indicates this solution can receive
+	// shared VPC selflinks (fully qualified compute links).
+	AllowSharedVpcs bool `yaml:"allowSharedVpcs"`
+	// Used to indicate to which machine type this network interface will be
+	// attached to.
+	MachineTypeVariable string `yaml:"machineTypeVariable"`
+}
+
+type ExternalIpType string
+
+const (
+	IpEphemeral ExternalIpType = "EPHEMERAL"
+	IpStaic     ExternalIpType = "STATIC"
+)
+
+type GceExternalIpExtension struct {
+	// NetworkVariable is used to indicate the network variable in the schema
+	// this external IP belongs to.
+	NetworkVariable string `yaml:"networkVariable"`
+
+	// Type specifies if the external IP is ephemeral or static.
+	// Defaults to ephemeral if not specified.
+	Type ExternalIpType `yaml:"externalIpType,omitempty"`
+}
+
+type GceIpForwardingExtension struct {
+	// NetworkVariable is used to indicate the network variable in the schema
+	// this external IP belongs to.
+	NetworkVariable string `yaml:"networkVariable"`
+	NotConfigurable bool   `yaml:"notConfigurable"`
+}
+
+type GceFirewallExtension struct {
+	// NetworkVariable is used to indicate the network variable in the schema
+	// this external IP belongs to.
+	NetworkVariable string `yaml:"networkVariable"`
+}
+
+type GceFirewallRangeExtension struct {
+	// FirewallVariable is used to indicate the firewall variable with the type
+	// GceFirewall in the schema to which this firewall range belongs to.
+	FirewallVariable string `yaml:"firewallVariable"`
+}
+
+type GceSubnetworkExtension struct {
+	// Subnetwork variable requires a network context in order to determine the
+	// set of available subnetworks. This field references another
+	// variable from the schema, which must have type GceNetwork.
+	NetworkVariable string `yaml:"networkVariable"`
+}
+
+type GceGenericResourceExtension struct {
+	// GCE resource type to be fetched. This field references another
+	// property from the schema, which must have type GceGenericResource.
+	ResourceVariable string `yaml:"resourceVariable"`
+}
+
+type IamServiceAccountExtension struct {
```

Comment:
Are these the roles for the deploying SA (which is already in standard) or is this for something else?

Reply:
They should be a same but sometimes this extension is used to create SAs with certain roles that are part of the solution. They can have different roles compared to what's required for provisioning.